### PR TITLE
Removes cycleFinderSkip and assigns tasks to groups.

### LINKED ISSUE #145: Clean up remaining taskSkip flags.
The correct way to disable a task by default is `enabled = false` in the task creation.

The correct way for a user to enable or disable a task permanently is
```
j2objcTASKNAME {
    enabled = true
}
```
or similar in their build.gradle.

The correct way for a user to disable a task temporarily is:
`./gradlew build -x TASKTOSKIP`

### PR DESCRIPTION
**Pull request by advayDev1**
Fixes #145.

### CODE REVIEW

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPlugin.groovy [hunk 1]
```diff
@@ -100,12 +100,18 @@ class J2objcPlugin implements Plugin<Project> {
             // on such generation.
             tasks.create(name: 'j2objcPreBuild', type: DefaultTask,
                     dependsOn: 'test') {
+                group 'build'
                 description "Marker task for all tasks that must be complete before j2objc building"
             }
 
+            // If you want to enable the j2objcCycleFinder task, use:
```
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPlugin.groovy]: Better formatting:

    // To enable the j2objcCycleFinder task, add the following to build.gradle:
    // j2objcCycleFinder { enabled = true }

**Comment by brunobowden** [comment]
I edited my comment twice, so check GitHub for the latest version.

What is the benefit of assigning to groups? The commit description and/or should have more explanation for this.

Looks good.

**Comment by confile** [comment]
@brunobowden I create some docs about that: https://github.com/j2objc-contrib/j2objc-gradle/pull/148

**Comment by brunobowden** [comment]
@advayDev1 - what's the purpose of assigning tasks to groups?

**Comment by advayDev1** [comment]
thanks @confile.

@brunobowden - finished comments, and modified commit to explain task groups.

**Pull request merged**